allennlp/training/trainer.py

```diff
@@ -514,6 +514,7 @@ def __init__(
 
         self._checkpointer: Optional[Checkpointer] = checkpointer
         if checkpointer is None and serialization_dir is not None:
+            os.makedirs(serialization_dir, exist_ok=True)
```

Comment:
So if `checkpointer` is not `None` we might still not create the serialization directory? I think we want to create it either way, right?